#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_new_gucharmap.dpatch by Steve Kowalik <stevenk@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Correct pkg-config name for gucharmap, and correct some code changes
## DP: with the new version of the library.

@DPATCH@
diff -urNad stardict-3.0.1~/configure.in stardict-3.0.1/configure.in
--- stardict-3.0.1~/configure.in	2008-09-01 18:37:55.000000000 +1000
+++ stardict-3.0.1/configure.in	2008-09-01 18:37:56.000000000 +1000
@@ -76,7 +76,7 @@
 	[enable_gucharmap=yes])
 
 if test "x$enable_gucharmap" = "xyes" ; then
-	PKG_CHECK_MODULES(GUCHARMAP, [gucharmap >= 1.4.0], have_gucharmap=yes, have_gucharmap=no)
+	PKG_CHECK_MODULES(GUCHARMAP, [gucharmap-2 >= 1.4.0], have_gucharmap=yes, have_gucharmap=no)
 	if test "x$have_gucharmap" = "xyes"; then
 		GUCHARMAP_CFLAGS="${GUCHARMAP_CFLAGS}"
 		GUCHARMAP_LIBS="${GUCHARMAP_LIBS}"
diff -urNad stardict-3.0.1~/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp stardict-3.0.1/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp
--- stardict-3.0.1~/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	2008-09-01 18:37:55.000000000 +1000
+++ stardict-3.0.1/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	2008-09-01 18:38:49.000000000 +1000
@@ -237,7 +237,7 @@
 	/* canonical decomposition */
 	gunichar *decomposition;
 	gsize result_len;
-	decomposition = gucharmap_unicode_canonical_decomposition (uc, &result_len);
+	decomposition = g_unicode_canonical_decomposition (uc, &result_len);
 	if (result_len != 1) {
 		definition += _("Canonical decomposition:");
 		definition += " ";
